### PR TITLE
feat: handle delete responses

### LINKED ISSUE #2703: Better handling delete responses to client
# Rationale

This is a part of #2396 to handle better delete responses. Since delete needs client-server interaction here and I was not clear how to handle them, I have left a few todos in the code. Now the process is clear, it is time to implement them. Search for `NGA todo` for those.

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #2703 
Now the delete process is clear, this PR handles the responses that were  unclear before

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 1]
```diff
@@ -494,5 +494,4 @@ message DeleteRequest {
 }
 
 message DeleteResponse {
-  // NGA todo: define an appropriate response
```
Comment (NGA-TRAN) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:497]: Follow all other similar action, this response is empty if successful

File: influxdb_iox_client/src/client/management.rs [hunk 2]
```diff
@@ -1057,7 +1057,6 @@ impl Client {
                 _ => DeleteError::ServerError(status),
             })?;
 
-        // NGA todo: return a handle to the delete?
```
Comment (NGA-TRAN) [inline influxdb_iox_client/src/client/management.rs:1060]: similarly, O(()) is the right return

File: src/influxdb_ioxd/rpc/error.rs [hunk 3]
```diff
@@ -177,6 +177,12 @@ pub fn default_db_error_handler(error: server::db::Error) -> tonic::Status {
             ),
         }
         .into(),
+        Error::DeleteFromTable { source, table_name } => PreconditionViolation {
+            category: "database".to_string(),
+            subject: "influxdata.com/iox".to_string(),
+            description: format!("Cannot delete data from table: {} : {}", table_name, source),
+        }
+        .into(),
```
Comment (NGA-TRAN) [inline src/influxdb_ioxd/rpc/error.rs:185]: This is the expected delete response when there is an error

File: tests/end_to_end_cases/management_api.rs [hunk 4]
```diff
@@ -1587,8 +1587,10 @@ async fn test_delete() {
     let pred = "region = west";
     let del = management_client
         .delete(db_name.clone(), table, start, stop, pred)
-        .await;
-    assert!(del.is_err());
+        .await
+        .unwrap_err()
+        .to_string();
+    assert!(del.contains("Cannot delete data from table"));
```
Comment (NGA-TRAN) [inline tests/end_to_end_cases/management_api.rs:1593]: capture error message at client

File: predicate/src/delete_predicate.rs [hunk 5]
```diff
@@ -705,7 +705,6 @@ mod tests {
         assert_eq!(expected, result);
     }
 
-    // NGA todo: check content of error messages
```
Comment (NGA-TRAN) [inline predicate/src/delete_predicate.rs:708]: Leftover comment. The error messages were checked in previous PR

File: src/influxdb_ioxd/http.rs [hunk 6]
```diff
@@ -282,7 +282,7 @@ impl ApplicationError {
             Self::ParsingLineProtocol { .. } => self.bad_request(),
             Self::ParsingDelete { .. } => self.bad_request(),
             Self::BuildingDeletePredicate { .. } => self.bad_request(),
-            Self::ExecutingDelete { .. } => self.internal_error(),
+            Self::ExecutingDelete { .. } => self.bad_request(),
```
Comment (NGA-TRAN) [inline src/influxdb_ioxd/http.rs:285]: Table not found is an example of error during execution. So it is a bad request.

File: src/influxdb_ioxd/http.rs [hunk 7]
```diff
@@ -1150,6 +1150,44 @@ mod tests {
...
+            .body(delete_line)
+            .send()
+            .await;
+        check_response(
+            "delete",
+            response,
+            StatusCode::BAD_REQUEST,
+            Some("Cannot delete data from non-existing table"),
+        )
+        .await;
```
Comment (NGA-TRAN) [inline src/influxdb_ioxd/http.rs:1190]: These are a part of delete  http endpoint tests

**Review by e-dard — APPROVED** [review]
LGTM 👍

**Pull request merged**